File: ApiRestRestaurant1/src/main/java/com/example/ApiRestRestaurant1/entities/Plat.java
package com.example.ApiRestRestaurant1.entities;

import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.FieldDefaults;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
public class Plat {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    Long platid;
    String nom;
    String description;
    Double prix;
    Double notemoyennesur10;
    String imageUrl; // Peut être null
    Long restaurantid;
    Long categorieplatid;
}

File: RestaurantProjectFront/src/app/models/plat.ts
import { Restaurant } from './restaurant';
import { CategoriePlat } from './categorieplat';

export class Plat {
    platid!: number;
    nom!: string;
    description!: string;
    prix!: number;
    notemoyennesur10!: number;
    imageUrl?: string; // Optional attribute
    restaurantid!: number; // Reference to the Restaurant model
    categorieplatid!: number; // Reference to the CategoriePlat model
}
